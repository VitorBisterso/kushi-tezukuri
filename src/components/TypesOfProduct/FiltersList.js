import React, { useState } from 'react'
import { useSelector } from 'react-redux'

import Checkbox from '../Checkbox'

const FiltersList = () => {
  const [filters, setFilters] = useState([])

  const { typesOfProduct } = useSelector(state => state.productsReducer)

  const handleCheckboxChange = (e, typeOfProduct) =>
    e.target.checked
      ? setFilters([...filters, typeOfProduct])
      : setFilters(filters.filter(f => f !== typeOfProduct))

  return typesOfProduct.map(type => {
    const { typeOfProduct, amount } = type
    const isChecked = filters.includes(typeOfProduct)

    return (
      <div key={typeOfProduct}>
        <Checkbox
          value={isChecked}
          onChange={e => handleCheckboxChange(e, typeOfProduct)}
        />
        <p data-testid="typeOfProductName">
          {typeOfProduct} ({amount})
        </p>
      </div>
    )
  })
}

export default FiltersList
