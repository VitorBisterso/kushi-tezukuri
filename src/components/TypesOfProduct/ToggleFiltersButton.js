import React from 'react'
import PropTypes from 'prop-types'

const ToggleFiltersButton = ({ isVisible, textAlign, onClick, title }) =>
  isVisible && (
    <button
      type="button"
      className="typesOfProduct-filtersButton"
      style={{ textAlign }}
      onClick={onClick}
    >
      {title}
    </button>
  )

ToggleFiltersButton.propTypes = {
  isVisible: PropTypes.bool,
  textAlign: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
  title: PropTypes.string.isRequired,
}

ToggleFiltersButton.defaultProps = {
  isVisible: true,
}

export default ToggleFiltersButton
