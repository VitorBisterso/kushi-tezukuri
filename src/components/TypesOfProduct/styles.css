.typesOfProduct-container {
  flex: 1;
  flex-direction: column;
}

.typesOfProduct-filtersHeader {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
}

.typesOfProduct-filtersHeader > p {
  margin-top: 0;
  font-size: var(--TEXT-MEDIUM);
  color: var(--COLOR-BLACK);
  font-family: 'Mali-SemiBold';
}

.typesOfProduct-filtersButton {
  display: none;
  cursor: pointer;
  border: none;
  background-color: var(--COLOR-WHITE);
  text-decoration: underline;
}

.typesOfProduct-filters {
  display: flex;
  flex-direction: column;
  padding: 10px;
  border-radius: var(--BORDER_RADIUS-REGULAR);
  box-shadow: 0px 5px 15px var(--COLOR-SHADOW);
}

.typesOfProduct-filters > div {
  display: flex;
  flex-direction: row;
  align-items: center;
}

.typesOfProduct-filters p {
  margin-left: 15px;
  font-size: var(--TEXT-REGULAR);
}

@media (max-width: 767px) {
  .typesOfProduct-container {
    width: 75%;
    margin: 0 auto 20px auto;
  }

  .typesOfProduct-filtersHeader > p {
    font-size: var(--TEXT-REGULAR);
  }

  .typesOfProduct-filtersButton {
    font-family: 'Mali-SemiBold';
    display: block;
    color: var(--COLOR-GOLD);
    margin-bottom: 10px;
  }
}
