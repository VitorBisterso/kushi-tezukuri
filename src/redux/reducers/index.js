import { combineReducers } from 'redux'
import productsReducer from './products'

const allReducers = combineReducers({
  productsReducer,
})

export default allReducers
