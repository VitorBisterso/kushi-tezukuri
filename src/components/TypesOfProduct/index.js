import React, { useState } from 'react'

import ToggleFiltersButton from './ToggleFiltersButton'
import FiltersList from './FiltersList'

import './styles.css'

const TypesOfProduct = () => {
  const [filtersVisibility, setFiltersVisibility] = useState(true)

  const handleButtonClick = () => setFiltersVisibility(!filtersVisibility)

  const renderToggleFiltersButton = visibility => (
    <ToggleFiltersButton
      isVisible={visibility}
      onClick={handleButtonClick}
      textAlign={filtersVisibility ? 'right' : 'center'}
      title={filtersVisibility ? 'Ocultar filtros' : 'Mostrar filtros'}
    />
  )

  return (
    <>
      {renderToggleFiltersButton(!filtersVisibility)}
      <div
        className="typesOfProduct-container"
        style={{ display: filtersVisibility ? 'flex' : 'none' }}
      >
        <div className="typesOfProduct-filtersHeader">
          <p>Filtrar por produto</p>
          {renderToggleFiltersButton(filtersVisibility)}
        </div>
        <div className="typesOfProduct-filters">
          <FiltersList />
        </div>
      </div>
    </>
  )
}

export default TypesOfProduct
