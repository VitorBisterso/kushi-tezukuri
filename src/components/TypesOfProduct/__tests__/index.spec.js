import React from 'react'
import { shallow } from 'enzyme'

import { render, waitFor } from '../../../redux/testUtils'

import TypesOfProduct from '../index'
import ToggleFiltersButton from '../ToggleFiltersButton'
import FiltersList from '../FiltersList'

describe('TypesOfProduct', () => {
  it('should render TypesOfProduct correctly', () => {
    const component = shallow(<TypesOfProduct />)

    expect(component).toMatchSnapshot()
  })

  it('should render ToggleFiltersButton correctly', () => {
    const clickFn = jest.fn()
    const component = shallow(
      <ToggleFiltersButton
        textAlign="center"
        onClick={clickFn}
        title="Test title"
      />
    )

    expect(component).toMatchSnapshot()
  })

  it('should render FiltersList correctly', async () => {
    const { getByTestId } = render(<FiltersList />, {
      initialState: {
        productsReducer: {
          typesOfProduct: [{ typeOfProduct: 'mask', amount: 1 }],
        },
      },
    })

    const textNode = await waitFor(() => getByTestId('typeOfProductName'))
    expect(textNode).toBeDefined()
  })
})
