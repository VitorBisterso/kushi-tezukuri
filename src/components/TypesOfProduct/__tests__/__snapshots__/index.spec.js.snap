// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TypesOfProduct should render ToggleFiltersButton correctly 1`] = `
<button
  className="typesOfProduct-filtersButton"
  onClick={[MockFunction]}
  style={
    Object {
      "textAlign": "center",
    }
  }
  type="button"
>
  Test title
</button>
`;

exports[`TypesOfProduct should render TypesOfProduct correctly 1`] = `
<Fragment>
  <ToggleFiltersButton
    isVisible={false}
    onClick={[Function]}
    textAlign="right"
    title="Ocultar filtros"
  />
  <div
    className="typesOfProduct-container"
    style={
      Object {
        "display": "flex",
      }
    }
  >
    <div
      className="typesOfProduct-filtersHeader"
    >
      <p>
        Filtrar por produto
      </p>
      <ToggleFiltersButton
        isVisible={true}
        onClick={[Function]}
        textAlign="right"
        title="Ocultar filtros"
      />
    </div>
    <div
      className="typesOfProduct-filters"
    >
      <FiltersList />
    </div>
  </div>
</Fragment>
`;
